pygridtools/iotools.py

```diff
@@ -233,3 +234,165 @@ def interactive_grid_shape(grid, max_n=200, plotfxn=None, **kwargs):
         plotfxn=ipywidgets.fixed(plotfxn),
         plotopts=ipywidgets.fixed(kwargs)
     )
+
+class _FocusProperties():
+    """A dummy class to hold the properties of the grid._FocusPoint() object.
+    This class is required so that multiple ipywidgets.interactive widgets
+    can interact on the same plot.
+    """
+    def __init__(self, pos=0.5, axis='x', factor=0.5, extent=0.5):
+        """Parameters
+        ----------
+        pos : float
+            Relative position within the grid of the focus. This must
+            be in the range [0, 1]
+        axis : string ('x' or 'y')
+            Axis along which the grid will be focused.
+        factor : float
+            Amount to focus grid. Creates cell sizes that are factor
+            smaller (factor > 1) or larger (factor < 1) in the focused
+            region.
+        extent : float
+            Lateral extent of focused region."""
+
+        self.pos=pos
+        self.axis=axis
+        self.factor=factor
+        self.extent=extent
+
+    @property
+    def focuspoint(self):
+        """Property returns grid._FocusPoint"""
+        return pyggrid._FocusPoint(pos=self.pos, axis=self.axis, factor=self.factor, extent=self.extent)
+
+def _plot_focus_points(focus_points, g, plotfxn, plotopts=None):
+    """Plots multiple focus points on a grid.
+    Parameters
+    ----------
+    focus_points : tuple of FocusProperties
+        These focus points are applied to grid `g`.
+    g : grid.Gridgen
+        The grid to plot.
```

Comment:
fill out the rest of the parameters here

Reply:
Will do